src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5565,6 +5625,44 @@ void SessionImpl::handleTorrentDeleteFailedAlert(const lt::torrent_delete_failed
     m_removingTorrents.erase(removingTorrentDataIter);
 }
 
+void SessionImpl::handleTorrentNeedCertAlert(const lt::torrent_need_cert_alert *p)
+{
+    const QString torrentID = TorrentID(p->handle.info_hash()).toString();
+
+    TorrentImpl *const torrent = m_torrents.value(p->handle.info_hash());
+    if (!torrent) [[unlikely]]
+        return;
+
+    const Path baseDir = sslPeersCertificatesDirectory();
+    if (baseDir.exists())
+    {
+
+        Path certPath = baseDir / Path(torrentID);
```

Comment:
Cert file has no extension unlike other two files. Is this done by intention?

Reply:
I didn't think much about it. I somehow just followed sub-conscientiously the fact of adding the suffix to the key file the same way haproxy does (https://docs.haproxy.org/2.6/configuration.html#5.1-crt). Do you think it's better to add a suffix? 

Maybe having `<name>.cert.pem`, `<name>.key.pem` and `<name.dh.pem>` would be better for operating systems where the suffix actually matters. I'm too used working on linux.

Reply:
> Maybe having `<name>.cert.pem`, `<name>.key.pem` and `<name>.dh.pem` would be better for operating systems where the suffix actually matters.

I would just use `.crt` for certificate files without adding the same `.pem` extension for all the files.

Reply:
> > Maybe having `<name>.cert.pem`, `<name>.key.pem` and `<name>.dh.pem` would be better for operating systems where the suffix actually matters.
> 
> I would just use `.crt` for certificate files without adding the same `.pem` extension for all the files.

Nevermind.
I'm OK with your suggestion. IIRC, `.pem` extension is often used for all such files.